executor/join.go

```diff
@@ -528,7 +681,22 @@ func (e *HashJoinExec) buildHashTableForList(buildSideResultCh <-chan *chunk.Chu
 		if e.finished.Load().(bool) {
 			return nil
 		}
-		err := e.rowContainer.PutChunk(chk)
+		if !e.useOuterToBuild {
+			err = e.rowContainer.PutChunk(chk)
+		} else {
+			var bitMap = bitmap.NewConcurrentBitmap(chk.NumRows())
+			e.outerMatchedStatus = append(e.outerMatchedStatus, bitMap)
+			e.memTracker.Consume(bitMap.BytesConsumed())
+			if len(e.outerFilter) > 0 {
```

Comment:
`if len(e.outerFilter) == 0` may be more readable.

Reply:
ok

Reply:
ok